blob/blob.go

```diff
@@ -522,6 +524,69 @@ type WriterOptions struct {
 	BeforeWrite func(asFunc func(interface{}) bool) error
 }
 
+// FromURLFn is for use by provider implementations.
+// It allows providers to convert a parsed URL from Open to a driver.Bucket.
+// It takes a bucket name and a map of key/value options. The map is guaranteed
+// to be non-nil, and all keys will be in lowercase.
+type FromURLFn func(context.Context, string, map[string]string) (driver.Bucket, error)
+
+// registry maps protocol strings to provider-specific instantiation functions.
+var registry = map[string]FromURLFn{}
+
+// Register is for use by provider implementations. It allows providers to
+// register an instantiation function for URLs with the given protocol. It is
+// expected to be called from the provider implementation's package init
+// function.
+//
+// fn will be called from Open, with a bucket name and options parsed from
+// the URL. All option keys will be lowercased.
+//
+// An error is returned if a provider has already registered for protocol.
```

Comment:
That leaves it up to the providers to check, and they may not be motivated to do so. I'd suggest panicking instead.

Reply:
Done.